src/python/alog/alog.py

```diff
@@ -577,6 +594,19 @@ def use_channel(channel):
     """
     return logging.getLogger(channel)
 
+## Logger Filters ##############################################################
```

Comment:
Hmmm, this is interesting. I have two reactions to this:

1. I don't think this is what `logging.Filter` is meant to do. I think [LogAdapter](https://docs.python.org/3/library/logging.html#loggeradapter-objects) is probably the more correct utility for this type of contextual information.
2. This makes me think that some of the `alog` functionality around filtering should be refactored to use `logging.Filter` in a more standard way

Reply:
I tried to use `LogAdapter`, however, but it modifies the `logger` which is read only inside `_ScopedLog`

Reply:
Filters are also designed / recommended for this use-case it seems: https://docs.python.org/3/howto/logging-cookbook.html#filters-contextual

Reply:
> This makes me think that some of the alog functionality around filtering should be refactored to use logging.Filter in a more standard way

I was thinking same while exploring adapters and filtering.

Reply:
Also current implementation isn't thread safe. I think for it to be threadsafe, I'll need to save this metadata in a `threadlng.local` based class object similar to how we do it for indent.

Reply:
Yeah, this is the nastiest part IMO. I think we'd want some careful threading behavior:

* Independent threads manage independent `metadata` contexts
* Spawned threads inherit context `metadata` from parents
* Closing a metadata scope removes the metadata from the thread where the scope was created
    * I'm honestly not sure what should happen to threads spawned by this parent thread while the scope was open

Also, the same behavior should _probably_ be true of spawned processes